### ISSUE TITLE
Wrong  Delineate Flow Oath

### ISSUE THREAD

**Issue by FernandaDavi** [opengeos/lidar#27]
* lidar version:
* Python version: 3.6.9
* Operating System: Windows x64-Conda

### Description
I am running the example you provide in the Lidar folder.
When I run script 3 I get the following error:

Traceback (most recent call last):
  File "C:\lidar\lidar\toolbox\scripts\3_Flow_Path.py", line 425, in <module>
    FlowPath(in_dem, in_sink, rain_intensity, out_flowpath)
  File "C:\lidar\lidar\toolbox\scripts\3_Flow_Path.py", line 303, in FlowPath
    code_block="",
  File "c:\program files\arcgis\pro\Resources\arcpy\arcpy\management.py", line 4530, in CalculateField
    raise e
  File "c:\program files\arcgis\pro\Resources\arcpy\arcpy\management.py", line 4527, in CalculateField
    retval = convertArcObjectToPythonObject(gp.CalculateField_management(*gp_fixargs((in_table, field, expression, expression_type, code_block, field_type), True)))
  File "c:\program files\arcgis\pro\Resources\arcpy\arcpy\geoprocessing\_base.py", line 511, in <lambda>
    return lambda *args: val(*gp_fixargs(args, True))
arcgisscripting.ExecuteError:  ERROR 000539:   File "<expression>", line 1
    0,05
       ^
SyntaxError: invalid token

Failed to execute (CalculateField).
 Failed to execute (DelineateFlowPath).

### What I Did

```
Paste the command(s) you ran and the output.
If there was a crash, please include the traceback here.
```
![Captura de pantalla (69)](https://user-images.githubusercontent.com/108424411/176725973-a7dc545e-3352-42ea-adae-42b4cdcd3179.png)
![Captura de pantalla (68)](https://user-images.githubusercontent.com/108424411/176725981-f34b50e7-c110-45c4-bf81-004f20a536a7.png)

**Comment by giswqs** [comment]
What ArcGIS Pro version are you using? Newer version of ArcGIS Pro might cause issues due to API changes.

**Comment by wobrotson** [comment]
I am experiencing a similar issue with Delineate Flowpath, though instead the error was an ExecuteError: [Error00059](https://pro.arcgis.com/en/pro-app/2.9/tool-reference/tool-errors-and-warnings/001001-010000/tool-errors-and-warnings-00526-00550-000539.htm). See below image.

![image](https://user-images.githubusercontent.com/40319079/191942942-c44f0772-a257-462f-ae60-da356adedd9f.png)

I'm running ArcPro 2.9.3. I'm not sure what other information you need from me?

Thanks for all your hard work!

**Comment by giswqs** [comment]
It seems the recent updates of ArcPy cause the issues. I will look into that next week when I have time. You can try out the toolbox ArcMap version for now.

**Comment by Dorka44** [comment]
Hi, first of all, thank you so much for this great tool for hydrological analysis! Like the others above, I've also encountered the same error - 000539. Furthermore, the subsequent steps don't seem to work either. My version of ArcGIS Pro is 3.1.3. Have you managed to resolve this issue by any chance?
Thank you in advance for any help!

**Comment by thompsongis** [comment]
I've jumped around from Arc 3.2.3 to 3.1.0 and now to 2.9.0 and I still am having issues with this step in the process getting the same error message as @FernandaDavi, @Dorka44 and @wobrotson. My traceback was exactly the same as @wobrotson. I was successful once with 3.1.0 but I can't process steps after #5. Has anyone had any success with this step since they've posted their issues? If so, what was your work-around? Unfortunately you can no longer download ArcMap and I regrettably have uninstalled that from my computer...

**Comment by thompsongis** [comment]
I figured out the issue! If your DEM has a LZ77 compression type or _potentially_ anything other than an LZW, you will run into issues with this step of the analysis. This is true for multiple versions of Pro and the packages used. To change this you must do so in a .gdb. 

I am still unable to complete step 6 & 7, but I'm thinking the solution may be one similar to this as the traceback has a similar structure/pattern.

**Comment by giswqs** [comment]
This issue was reported with Python 3.6 which is no longer supported. If you can reproduce this with a current Python version (3.9+) and the latest lidar package, please open a new issue. Closing as stale.
